src/Parser/RecursiveTextProcessor.php

```diff
@@ -136,11 +136,7 @@ public function transcludeAnnotation( $transcludeAnnotation ) {
 		}
 
 		$parserOutput = $this->parser->getOutput();
-		$track = $parserOutput->getExtensionData( ParserData::ANNOTATION_BLOCK );
-
-		if ( $track === null ) {
-			$track = [];
-		}
+		$track = $parserOutput->getExtensionData( ParserData::ANNOTATION_BLOCK ) ?: [];
```

Comment:
This should be `??`

Reply:
At least to retain the exact same behavior as the old code

Reply:
Unfortunately that's not sufficient because this property may be set to `false` explicitly—some other places e.g. L313 already check for that. On PHP < 8.1, this causes the var to be silently converted into an array when L145 accesses a property on it, which triggers a warning on PHP >= 8.1. So changing it to `?:` just makes that behavior explicit.